iris/src/iris/llm/external/ollama.py

```diff
@@ -148,25 +153,204 @@ def complete(
         )
         return response["response"]
 
+    def _callable_to_ollama_tool(self, func: callable) -> Dict[str, Any]:
+        """Create a minimal OpenAI-style tool schema for a plain callable."""
+        params = {"type": "object", "properties": {}}
+        sig = inspect.signature(func)
+        # If the callable actually takes arguments, expose them as free-form
+        # (customize here to generate a real JSON Schema if you like)
+        if any(
+            p.kind in (p.POSITIONAL_OR_KEYWORD, p.KEYWORD_ONLY)
+            for p in sig.parameters.values()
+        ):
+            params = {"type": "object", "additionalProperties": True}
+        return {
+            "name": func.__name__,
+            "description": (func.__doc__ or "").strip()
+            or f"Callable tool {func.__name__}",
+            "parameters": params,
+        }
```

Comment:
The Ollama Python API already has a functionality to convert Callables to proper Tool objects that it needs. See for example the documentation of the chat method in the ollama client:
```
    Create a chat response using the requested model.

    Args:
      tools:
        A JSON schema as a dict, an Ollama Tool or a Python Function.
        Python functions need to follow Google style docstrings to be converted to an Ollama Tool.
        For more information, see: https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings
```
I would recommend to just pass Callables to the Ollama Client directly (or use Ollamas integrated functions) as they are guaranteed to always convert them to the best format for Ollama.